### PR TITLE
Handle cpu dtype requirements for conv

### PR DESCRIPTION
**Pull request by EricLBuehler**

### CODE REVIEW

**Comment by sempervictus** [comment]
@EricLBuehler - is this to address
```rust
2025-10-11T11:26:57.744803Z  INFO mistralrs_core::prefix_cacher: PrefixCacherV2 is enabled. Expect higher multi-turn throughput for both text and multimodal.
2025-10-11T11:26:57.972149Z ERROR mistralrs_core::engine: prompt step - Model failed with error: Expected CPU storage for k
2025-10-11T11:26:57.972646Z  INFO mistralrs_core: Dummy run completed in 0.234968657s.
2025-10-11T11:26:57.977757Z  INFO mistralrs_server: OpenAI-compatible server listening on http://0.0.0.0:12099.
^C
```

**Comment by EricLBuehler** [comment]
Hi @sempervictus! Can you please give more details about `Expected CPU storage for k`

**Comment by sempervictus** [comment]
Those were all that i caught, will do a rebuild/retest over the weekend now that this is merged

**Comment by sempervictus** [comment]
@EricLBuehler 

> 2025-10-18T00:07:25.872909Z  INFO mistralrs_quant::utils::log: Layers 0-34: cuda[0] (16 GB)
> 2025-10-18T00:07:25.906737Z  INFO mistralrs_core::utils::normal: Detected minimum CUDA compute capability 8.9
> 2025-10-18T00:07:25.907107Z  INFO mistralrs_core::utils::normal: DType selected is BF16.
> 2025-10-18T00:07:25.907490Z  INFO mistralrs_core::pipeline::vision: Model config: Gemma3nConfig { text_config: Gemma3nTextConfig { attention_bias: false, head_dim: 256, hidden_activation: GeluPytorchTanh, hidden_size: 2048, intermediate_size: PerLayer([16384, 16384, 16384, 16384, 16384, 16384, 16384, 16384, 16384, 16384, 16384, 16384, 16384, 16384, 16384, 16384, 16384, 16384, 16384, 16384, 16384, 16384, 16384, 16384, 16384, 16384, 16384, 16384, 16384, 16384, 16384, 16384, 16384, 16384, 16384]), num_attention_heads: 8, num_hidden_layers: 35, num_kv_shared_layers: 15, num_key_value_heads: 2, rms_norm_eps: 1e-6, rope_theta: 1000000.0, vocab_size: 262400, sliding_window: 512, max_position_embeddings: 32768, quantization_config: None, tie_word_embeddings: true, rope_local_base_freq: 10000.0, sliding_window_pattern: 6, rope_scaling: None, vocab_size_per_layer_input: 262144, hidden_size_per_layer_input: 256, altup_num_inputs: 4, layer_types: ["sliding_attention", "sliding_attention", "sliding_attention", "sliding_attention", "full_attention", "sliding_attention", "sliding_attention", "sliding_attention", "sliding_attention", "full_attention", "sliding_attention", "sliding_attention", "sliding_attention", "sliding_attention", "full_attention", "sliding_attention", "sliding_attention", "sliding_attention", "sliding_attention", "full_attention", "sliding_attention", "sliding_attention", "sliding_attention", "sliding_attention", "full_attention", "sliding_attention", "sliding_attention", "sliding_attention", "sliding_attention", "full_attention", "sliding_attention", "sliding_attention", "sliding_attention", "sliding_attention", "full_attention"], altup_active_idx: 0, altup_coef_clip: Some(120.0), laurel_rank: 64, altup_correct_scale: true, activation_sparsity_pattern: [0.95, 0.95, 0.95, 0.95, 0.95, 0.95, 0.95, 0.95, 0.95, 0.95, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0], final_logit_softcapping: Some(30.0) }, vision_config: Gemma3nVisionConfig { hidden_size: 2048, vocab_offset: 262144, vocab_size: 128 }, audio_config: Gemma3nAudioConfig { input_feat_size: 128, hidden_size: 1536, conf_attention_chunk_size: 12, conf_attention_context_left: 13, conf_attention_context_right: 0, conf_attention_invalid_logits_value: -1000000000.0, conf_attention_logit_cap: 50.0, conf_num_attention_heads: 8, conf_num_hidden_layers: 12, conf_conv_kernel_size: 5, conf_reduction_factor: 4, conf_residual_weight: 0.5, sscp_conv_channel_size: [128, 32], sscp_conv_kernel_size: [[3, 3], [3, 3]], sscp_conv_stride_size: [[2, 2], [2, 2]], vocab_size: 128, sscp_conv_group_norm_eps: 0.001, rms_norm_eps: 1e-6, vocab_offset: 262272 }, audio_soft_tokens_per_image: 188 }
> 2025-10-18T00:07:25.908411Z  INFO mistralrs_quant::utils::log: FlashAttention is enabled.
> 2025-10-18T00:07:25.908753Z  INFO mistralrs_core::utils::varbuilder_utils: Loading model using mmap strategy.
> 2025-10-18T00:08:50.953805Z  INFO mistralrs_core::pipeline::paths: Using chat template from .jinja file.
> 2025-10-18T00:08:50.954277Z  INFO mistralrs_core::pipeline::isq: Applying in-situ quantization into Some(Q8_0) to 388 tensors.
> 2025-10-18T00:08:50.954424Z  INFO mistralrs_core::pipeline::isq: Applying ISQ on 32 threads.
> 2025-10-18T00:08:58.505626Z  INFO mistralrs_core::pipeline::isq: Applied in-situ quantization into Some(Q8_0) to 388 tensors out of 388 total tensors. Took 7.55s
> 2025-10-18T00:08:59.544926Z  INFO mistralrs_core::pipeline::chat_template: bos_toks = "<bos>", eos_toks = "<end_of_turn>", "<eos>", unk_tok = `None`
> 2025-10-18T00:08:59.604176Z  INFO mistralrs_server_core::mistralrs_for_server_builder: Model loaded.
> 2025-10-18T00:08:59.605170Z  INFO mistralrs_core: Pipeline input modalities are [📝 Text, 🖼️ Vision, 🔊 Audio]
> 2025-10-18T00:08:59.605348Z  INFO mistralrs_core: Pipeline output modalities are [📝 Text]
> 2025-10-18T00:08:59.605602Z  INFO mistralrs_core: Beginning dummy run.
> 2025-10-18T00:08:59.610472Z  INFO mistralrs_core::prefix_cacher: PrefixCacherV2 is enabled. Expect higher multi-turn throughput for both text and multimodal.
> 2025-10-18T00:09:05.878384Z ERROR mistralrs_core::engine: prompt step - Model failed with error: narrow invalid args start + len > dim_len: [1, 1], dim: 1, start: 0, len:2
> 2025-10-18T00:09:05.879008Z  INFO mistralrs_core: Dummy run completed in 6.27340053s.
> 2025-10-18T00:09:05.885119Z  INFO mistralrs_server: OpenAI-compatible server listening on http://0.0.0.0:12099.

when run as `... --prefix-cache-n 8 --pa-cache-type f8e4m3 --pa-ctxt-len 16384 --isq q8_0 run -m google/gemma-3n-E4B-it --max-seq-len 3276`
